### PR TITLE
Documentation for remaining configuration options

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Document all configuration options.

## What

Documentation of options from `environment_variables.h`

## Tests

n/a

### CODE REVIEW

File: docs/advanced-config.md [hunk 1]
```diff
@@ -84,4 +84,5 @@ Use these environment variables to configure the tracing library:
 | `SIGNALFX_INSTRUMENTATION_ELASTICSEARCH_TAG_QUERIES` | Enable the tagging of a Elasticsearch command PostData as `db.statement`. It may introduce overhead for direct streaming users. | `true` |
 | `SIGNALFX_TRACE_DELAY_WCF_INSTRUMENTATION_ENABLED` | Enable the updated WCF instrumentation that delays execution until later in the WCF pipeline when the WCF server exception handling is established. | `false` |
 | `SIGNALFX_TRACE_KAFKA_CREATE_CONSUMER_SCOPE_ENABLED` | Enable to close consumer scope on method enter, and start a new one on method exit. | `true` |
-| `SIGNALFX_TRACE_{0}_ENABLED` | Configuration pattern for enabling/disabling a specific library instrumentation, e.g in order to disable Kafka instrumentation, set `SIGNALFX_TRACE_Kafka_ENABLED=false` | `true` |
\ No newline at end of file
+| `SIGNALFX_TRACE_{0}_ENABLED` | Configuration pattern for enabling/disabling a specific library instrumentation, e.g in order to disable Kafka instrumentation, set `SIGNALFX_TRACE_Kafka_ENABLED=false` | `true` |
```
Comment (theletterf) [inline docs/advanced-config.md:87]:
```suggestion
| `SIGNALFX_TRACE_{0}_ENABLED` | Configuration pattern for enabling or disabling a specific library instrumentation. For example, in order to disable Kafka instrumentation, set `SIGNALFX_TRACE_Kafka_ENABLED=false` | `true` |
```

File: docs/advanced-config.md [hunk 2]
```diff
@@ -84,4 +84,5 @@ Use these environment variables to configure the tracing library:
...
+| `SIGNALFX_AZURE_APP_SERVICES` | Set to indicate the profiler is running in the context of Azure App Services. | `false` |
```
Comment (theletterf) [inline docs/advanced-config.md:88]:
```suggestion
| `SIGNALFX_AZURE_APP_SERVICES` | Set to indicate that the profiler is running in the context of Azure App Services. | `false` |
```

File: docs/internal-config.md [hunk 3]
```diff
@@ -9,6 +9,10 @@ These settings should be never used by the users.
 |-|-|-|
 | `SIGNALFX_EXPORTER` | The exporter to be used. The Tracer uses it to encode and dispatch traces. Available values are: `DatadogAgent`, `Zipkin`. | `Zipkin` |
 | `SIGNALFX_CONVENTION` | Sets the semantic and trace id conventions for the tracer. Available values are: `Datadog` (64bit trace id), `OpenTelemetry` (128 bit trace id). | `OpenTelemetry` |
+| `SIGNALFX_DUMP_ILREWRITE_ENABLED` | Enable the profiler to dump the IL original code and modification to the log. | `false` |
```
Comment (theletterf) [inline docs/internal-config.md:12]:
```suggestion
| `SIGNALFX_DUMP_ILREWRITE_ENABLED` | Allows the profiler to dump the IL original code and modification to the log. | `false` |
```

File: docs/internal-config.md [hunk 4]
```diff
@@ -50,3 +53,4 @@ These settings should be never used by the users.
 | `SIGNALFX_METRICS_EXPORTER` | Metrics exporter to be used. It is used to encode and dispatch metrics. Available values are: `SignalFx`, `StatsD`. | `SignalFx` |
 | `SIGNALFX_THREAD_SAMPLING_ENABLED` | Enable to activate thread sampling. | `false` |
 | `SIGNALFX_THREAD_SAMPLING_PERIOD` | Sampling period. It defines how often the threads are stopped in order to fetch all stack traces. This value cannot be lower than `1000` milliseconds. | `1000` |
+| `SIGNALFX_TRACE_AZURE_FUNCTIONS_ENABLED` | Set to instrument within azure functions. | `false` |
```
Comment (theletterf) [inline docs/internal-config.md:56]:
```suggestion
| `SIGNALFX_TRACE_AZURE_FUNCTIONS_ENABLED` | Set to instrument within Azure functions. | `false` |
```

**Review by theletterf — CHANGES_REQUESTED** [review]

File: docs/advanced-config.md (hunk 2 above)
Comment (pellared) [inline docs/advanced-config.md:88]: Maybe we should move it to `internal-config.md`? Here is why: https://github.com/DataDog/dd-trace-dotnet/blob/ef56770ed22a53110a0ae987b283cb8ad30139c7/tracer/src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp#L144-L145

**Review by Kielek — APPROVED** [review]

**Review by theletterf — APPROVED** [review]

**Pull request merged**